dotnet/src/Skills/Skills.OpenAPI/Extensions/RestApiOperationExtensions.cs

```diff
@@ -66,6 +61,127 @@ public static IReadOnlyList<RestApiOperationParameter> GetParameters(this RestAp
         return parameters;
     }
 
+    /// <summary>
+    /// Retrieves the payload parameters for a given REST API operation.
+    /// </summary>
+    /// <param name="operation">The REST API operation to retrieve parameters for.</param>
+    /// <param name="useParametersFromMetadata">Flag indicating whether to include parameters from metadata.
+    /// If false or not specified, the 'payload' and 'content-type' parameters are added instead.</param>
+    /// <param name="namespacePayloadParamNames">Flag indicating whether to namespace payload parameter names.</param>
+    /// <returns>A list of <see cref="RestApiOperationParameter"/> representing the payload parameters.</returns>
+    private static List<RestApiOperationParameter> GetPayloadParameters(RestApiOperation operation, bool useParametersFromMetadata, bool namespacePayloadParamNames)
+    {
+        if (useParametersFromMetadata is true)
+        {
+            if (operation.Payload is null)
+            {
+                throw new SKException($"Payload parameters cannot be retrieved from the '{operation.Id}' operation payload metadata because it is missing.");
+            }
+
+            // The 'text/plain' content type payload metadata does not contain parameter names.
+            // So, returning artificial 'payload' parameter instead.
+            if (operation.Payload.MediaType == MediaTypeTextPlain)
+            {
+                return new List<RestApiOperationParameter> { CreatePayloadArtificialParameter(operation) };
+            }
+
+            return GetParametersFromPayloadMetadata(operation.Payload, namespacePayloadParamNames);
+        }
+
+        //Adding artificial 'payload' and 'content-type' in case parameters from payload metadata are not required.
+        return new List<RestApiOperationParameter> {
+            CreatePayloadArtificialParameter(operation),
+            CreateContentTypeArtificialParameter(operation)
+        };
+    }
+
+    /// <summary>
+    /// Creates artificial 'content-type' parameter for a REST API operation.
```

Comment:
Small update to keep XML documentation consistent with similar method for `payload` parameter and method name.
```suggestion
    /// Creates the 'content-type' artificial parameter for a REST API operation.
```